[WIP] tool: unify config files

jingyih opened the pull request:
Record "ignored fields" for direct resources in their config files, ensuring these files serve as the source of truth for ignored fields across all controller builder tools.

jingyih commented:
/cc @jasonvigil 

We discussed offline about the resource metadata.

justinsb commented:
I think this PR accidentally checks in a binary that should be .gitignored [dev/tools/controllerbuilder/controllerbuilder](https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/3669/files#diff-931cd848d0252142784da9d09a3e1d47e2f33efa3b2ce2000ba58e9c3753e656)

jingyih commented:
> I think this PR accidentally checks in a binary that should be .gitignored [dev/tools/controllerbuilder/controllerbuilder](https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/3669/files#diff-931cd848d0252142784da9d09a3e1d47e2f33efa3b2ce2000ba58e9c3753e656)

Thanks! Rebased and added the controllerbuilder binary to .gitignored
